### PR TITLE
Try to advise about core.useBuiltinFSMonitor's deprecation only once

### PR DESCRIPTION
**Pull request by dscho**
I noticed this a few times but was not able to come up with an easy way to trigger it, but I _think_ the code is obviously correct 😄

### CODE REVIEW

File: fsmonitor-settings.c [hunk 1]
```diff
@@ -59,9 +59,12 @@ static int check_deprecated_builtin_config(struct repository *r)
...
 	if(!repo_config_get_bool(r, "core.useBuiltinFSMonitor", &core_use_builtin_fsmonitor)) {
-		advise_if_enabled(ADVICE_USE_CORE_FSMONITOR_CONFIG,
-				  _("core.useBuiltinFSMonitor will be deprecated "
-				    "soon; use core.fsmonitor instead"));
+		if (!git_env_bool("GIT_SUPPRESS_USEBUILTINFSMONITOR_ADVICE", 0)) {
+			advise_if_enabled(ADVICE_USE_CORE_FSMONITOR_CONFIG,
+					  _("core.useBuiltinFSMonitor will be deprecated "
+					    "soon; use core.fsmonitor instead"));
+			setenv("GIT_SUPPRESS_USEBUILTINFSMONITOR_ADVICE", "1", 1);
+		}
```
Comment (derrickstolee) [inline fsmonitor-settings.c:67]: Why use the environment when a `static int` suffices?

```suggestion
		static int gave_advice = 0;
		if (!gave_advice) {
			advise_if_enabled(ADVICE_USE_CORE_FSMONITOR_CONFIG,
					  _("core.useBuiltinFSMonitor will be deprecated "
					    "soon; use core.fsmonitor instead"));
			gave_advice = 1;
		}
```

Comment (dscho) [reply fsmonitor-settings.c:67]: I was thinking about child processes, e.g. clean/smudge or diff filters, or hooks. There, a `static` won't suffice.

**Comment by derrickstolee** [comment]
> i'm wondering if the advise is even necessary. and certainly efforts to only see the message once are perhaps way too complicated. the core.useBuiltinFSMonitor was only present in experimental builds and we've always said that things may change before they finalize.

I think it would be helpful for Git for Windows users who set this manually. This would also help us over in `microsoft/git` to find out if `scalar register` is correctly overwriting the option with `core.fsmonitor=true` (as we will get user feedback complaining about the advice).

**Review by derrickstolee — APPROVED** [review]

**Comment by dscho** [comment]
> the core.useBuiltinFSMonitor was only present in experimental builds and we've always said that things may change before they finalize.

While that is true, I want to show our appreciation for early adopters and tire-kickers by letting them know that the config setting's name has changed, without asking them to RTFM.

To that end, we want to reinstate support for the old name (it still does what it used to do), and tell the user that they will have to change their configuration. We can keep doing this in v2.36, and drop it in v2.37.

@jeffhostetler could you agree with that approach?

**Review by jeffhostetler — APPROVED** [review]

**Pull request merged**